src/plugins/vis_type_vega/public/vega_view/vega_base_view.js

```diff
@@ -127,6 +127,46 @@ export class VegaBaseView {
     }
   }
 
+  /**
+   * Find index pattern by its title, of if not given, gets default
+   * @param {string} [index]
+   * @returns {Promise<string>} index id
+   */
+  async findIndex(index) {
+    const { indexPatterns } = getData();
+    let idxObj;
+
+    if (index) {
+      idxObj = await indexPatterns.findByTitle(index);
+      if (!idxObj) {
+        throw new Error(
+          i18n.translate('visTypeVega.visualization.indexNotFoundErrorMessage', {
+            defaultMessage: 'Index {index} not found',
+            values: { index: `"${index}"` },
+          })
+        );
+      }
+    } else {
+      [idxObj] = await extractIndexPatternsFromSpec(
+        this._parser.isVegaLite ? this._parser.vlspec : this._parser.spec
+      );
+
+      if (!idxObj) {
+        idxObj = await indexPatterns.getDefault();
+      }
```

Comment:
Can we combine these two ifs? It is not so readable right now.

Reply:
I'll will try to think how to make it more readable but I need both if's. Let me explain you why we need it:
1) we are trying to retrieve index pattern from Vega expression function: 
![image](https://user-images.githubusercontent.com/20072247/100340202-e45c3400-2feb-11eb-972f-a10fcc111f52.png)

2) if the user does not pass the index from the `expression`, we try to extract the index from the `spec`
![image](https://user-images.githubusercontent.com/20072247/100340256-f6d66d80-2feb-11eb-9286-c4d577a22d57.png)

3) if from we cannot extract index from spec we are using a default one.
![image](https://user-images.githubusercontent.com/20072247/100340381-24bbb200-2fec-11eb-8857-0909253de533.png)

4) if we cannot get a default index (I have no idea how it's possiblt,  but that logic wasn't added by me) we show exception
![image](https://user-images.githubusercontent.com/20072247/100340478-52086000-2fec-11eb-910e-6174bc4b9dad.png)

Reply:
Yy, I get why you are doing this but I don't like the two ifs tbh. If we do something like:
```
      if (!idxObj) {
        const defaultIdx = await indexPatterns.getDefault();
        if (defaultIdx) {
           idxObj = defaultIdx
       } else {
           throw new Error(
          i18n.translate('visTypeVega.visualization.unableToFindDefaultIndexErrorMessage', {
            defaultMessage: 'Unable to find default index',
          })
        );
      }
    }
```

Reply:
np =)